ResetPassword.js

```diff
@@ -40,6 +55,78 @@ module.exports = function (Model, options) {
                 });
             });
         });
+    /**
+      *  The following route is meant to generate OTP for valid users.
+      */
+        Model.app.post('/request-password-reset-phone', function(request, response, next) {
+            /*
+             * Check whether a user with the provided phone number exits.
+             * Phone numbers which don't correspond to any user are cons-
+             * dered unauthorized.
+             */
+            Model.findOne({ where: {
+                phone: request.body.phone
+            }}, (err, user) => {
+                if(!user)
+                    response.status(401).send({
+                        statusCode:401,
+                        message:'Invalid Phone Number'
+                    });
+                else {
+                    /*
+                     * Generate OTP by accessing the thrid party library.
+                     */ 
+                    sendOtp.send(request.body.phone, "PRIIND", function (error, data, result) {
+                        if(data.type === 'success') {
+                            response.send(true);
+                        }
+                        if(data.type == 'error') {
+                            response.status(504).send({
+                                "message":'OTP generation failed'
+                            });
+                        }
+                    });
+                }
+            });
+        });
+    /**
+      * The following route is meant for OTP verification.
+      */
+         Model.app.post('/verify-otp', function(request, response, next) {
+             let phone = request.body.phone;
+             let otp = request.body.otp;
+             sendOtp.verify(phone, otp, function (error, data, result) {
```

Comment:
Do we rely on the `sendOtp.verify` to check that the `phone` and `otp` values exist and are valid?

Reply:
Yes, the `verify` method of the `sendotp` module does the work of validating.